### ISSUE TITLE
[HiDPI] images created with SWT.IMAGE_GRAY looks zoomed in

### ISSUE THREAD

**Issue by dkartal** [eclipse-platform/eclipse.platform.swt#528]
**Describe the bug**
creating an image with flags SWT.IMAGE_GRAY, or IMAGE_DISABLE somehow changes the look of the image on high resolution screens on ubuntu. Please see org.eclipse.swt.graphics.Image#Image(org.eclipse.swt.graphics.Device, org.eclipse.swt.graphics.Image, int)

**To Reproduce**
```
public static void main(String[] args) {
  Display display = new Display();
  Shell shell = new Shell(display);
  shell.setLayout(new FillLayout());
  Image image1 = new Image(display, someLocalImage);
  Image image2 = new Image(display, image1, SWT.IMAGE_GRAY);
  Button button = new Button(shell, SWT.NONE);
  button.setImage(image1);
  button.addMouseListener(new MouseAdapter() {
    @Override
    public void mouseDown(MouseEvent e) {
      if (button.getImage().equals(image1)) {
        button.setImage(image2);
      }
      else {
        button.setImage(image1);
      }
    }
  });
  shell.open();
  while (!shell.isDisposed()) {
    if (!display.readAndDispatch()) {
      display.sleep();
    }
  }
  display.dispose();
}
```
image of the button looks good at first, (see the first image)
press the button and see the gray version of the image (see the second image)

**Expected behavior**
the image created with the SWT.IMAGE_GRAY flag should look as it is.

**Screenshots**
![Screenshot from 2023-01-16 19-18-16](https://user-images.githubusercontent.com/120656793/212743950-48910623-fb2f-4f52-9eb8-c6cc729d752c.png)

![Screenshot from 2023-01-16 19-17-51](https://user-images.githubusercontent.com/120656793/212744012-4142c46a-492a-4dc8-80bf-30be2cf7607d.png)

**Environment:**
1. Select the platform(s) on which the behavior is seen:
- - [ ] Windows
- - [x] Linux
- - [ ] macOS 

4. Additional OS info (e.g. OS version, Linux Desktop, etc)
Ubuntu 22.04

5. JRE/JDK version
 Java 17

**Comment by peterseverin** [comment]
What a coincidence, this is exactly the problem that I wanted to report. Derived images do not scale correctly when painted on Linux. Here's my snippet that uses a system image:

```
package org.eclipse.swt.snippets;

import org.eclipse.swt.*;
import org.eclipse.swt.graphics.*;
import org.eclipse.swt.widgets.*;

public class DerivedImageScaledPaintBug {
	public static void main (String [] args) {
		Display display = new Display ();
		Shell shell = new Shell (display, SWT.SHELL_TRIM | SWT.DOUBLE_BUFFERED);
		shell.setText("");
		final Image image = display.getSystemImage (SWT.ICON_QUESTION);
		final Image grayImage = new Image(display, image, SWT.IMAGE_GRAY);
		shell.addListener (SWT.Paint, e -> {
			Rectangle rect = image.getBounds ();
			int width = rect.width;
			int height = rect.height;
			GC gc = e.gc;
			int x = 10, y = 10;
			gc.drawImage (image, x, y);
			gc.drawImage (grayImage, x + width + 10, y);
		});
		shell.setSize (600, 400);
		shell.open ();
		while (!shell.isDisposed ()) {
			if (!display.readAndDispatch ())
				display.sleep ();
		}
		display.dispose ();
	}
}

```

It paints a normal image and the derived image on a HIDPI monitor. The derived image is painted as having half the size of the original image. The result looks like this:

![derived-image-paint-bug](https://user-images.githubusercontent.com/111872/213176487-88ac92c9-0c15-4603-9ad9-e7e38208cfdf.png)

On platforms other than Linux, or with x1 scale, both images have the same size.

**Comment by iloveeclipse** [comment]
@peterseverin : thanks for example, unfortunately I don't have such a monitor to see the problem.
I remember there was a trick with setting a variable to "mimic" HiDPI environment on "low" DPI monitor?

Is this a recent regression? If yes, which SWT verison is "OK"?

**Comment by peterseverin** [comment]
II use a Debian VM which I configure with x2 scaling, without using HiDPI monitor. But I believe that if you are using Linux with Gnome then you should be able to set x2 scale using System Settings > Displays.

**Comment by SyntevoAlex** [comment]
Right, 200% dpi can be enabled in Linux even on a regular monitor. I do it myself for HiDpi testing.

**Comment by iloveeclipse** [comment]
OK, I'm on (ancient) KDE 4.14 / RHEL 7.9 and don't immediately see a UI preference for that. Any shell variables one can use?

**Comment by peterseverin** [comment]
These versions seem old. I would use a VirtualBox VM to test HiDPI with a more recent version.

**Comment by iloveeclipse** [comment]
> These versions seem old. 

Yes, that is sad, but that is nothing I can/will change. Crossing fingers by the end of the year we will get RHEL 9.

> I would use a VirtualBox VM to test HiDPI with a more recent version.

I would too, if I would have time :-) Since HiDPI is not an issue for us (yet), it is not a high prio for me, sorry.

**Comment by Phillipus** [comment]
> I remember there was a trick with setting a variable to "mimic" HiDPI environment on "low" DPI monitor?

Set an Environment variable:

`GDK_SCALE=2`

**Comment by Phillipus** [comment]
This affects disabled menu and toolbar icons if a specified disabled image is not set.

**Comment by HeikoKlare** [comment]
This has been fixed via:
- https://github.com/eclipse-platform/eclipse.platform.swt/pull/2653
